reasoner/processor/reactive/AbstractStream.java

```diff
@@ -63,8 +63,10 @@ void propagatePull(Publisher<INPUT> publisher) {
 
     @Override
     public void registerPublisher(Publisher<INPUT> publisher) {
-        publisherRegistry().add(publisher);
-        subscriberDelegate().registerPath(publisher);
+        if (!publisherRegistry.contains(publisher)) {
+            publisherRegistry().add(publisher);
+            subscriberDelegate().registerPath(publisher);
+        }
```

Comment:
Prevents the duplicate edges between the same two streams. 

This shouldn't happen with the optimised conjunction stream structure (but might with explanations disabling the optimisation) and I should look into it and reply here on why/when it happens or add/replace it with an assert.